pkg/providers/tinkerbell/upgrade.go

```diff
@@ -146,7 +146,8 @@ func (p *Provider) validateAvailableHardwareForUpgrade(ctx context.Context, curr
 	)
 
 	rollingUpgrade := false
-	if currentSpec.Cluster.Spec.KubernetesVersion != newClusterSpec.Cluster.Spec.KubernetesVersion {
+	if currentSpec.Cluster.Spec.KubernetesVersion != newClusterSpec.Cluster.Spec.KubernetesVersion ||
+		currentSpec.Bundles.Spec.Number != newClusterSpec.Bundles.Spec.Number {
```

Comment:
will there always be a current bundle and new bundle?

Reply:
yeah, this codepath is called only for upgrade flow where the `currentSpec.Bundles` will be the old spec's bundle and the `newClusterSpec.Bundles` will be the new spec's bundle.